src/test/java/com/google/devtools/build/lib/packages/RuleClassTest.java

```diff
@@ -1109,4 +1083,48 @@ public void testExecutionPlatformConstraints_perTarget() throws Exception {
         .isEqualTo(ExecutionPlatformConstraintsAllowed.PER_TARGET);
     assertThat(ruleClass.hasAttr("exec_compatible_with", LABEL_LIST)).isTrue();
   }
+
+  @Test
+  public void testExecutionPlatformConstraints_inheritConstraintsFromParent() throws Exception {
+    RuleClass parentRuleClass =
+        new RuleClass.Builder("$parentRuleClass", RuleClassType.ABSTRACT, false)
+            .add(attr("tags", STRING_LIST))
+            .executionPlatformConstraintsAllowed(ExecutionPlatformConstraintsAllowed.PER_RULE)
+            .addExecutionPlatformConstraints(
+                Label.parseAbsolute("//constraints:cv1"), Label.parseAbsolute("//constraints:cv2"))
+            .build();
+
+    RuleClass.Builder childRuleClassBuilder =
```

Comment:
Nit: This builder looks like it can be inlined.

Reply:
Done.